mesh/profiles/views.py

```diff
@@ -44,3 +60,169 @@ def profile_picture(request):
+                                    status = 409)
+            
+            imgur_client = initialize_imgur_client()
+            
+            # Grab raw photo from request files and encode it to upload to imgur
+            image_file = b64encode(request.FILES["profilePicture"].read())
+
+            # send POST request to imgur API to upload photo
+            imgur_response = imgur_client._send_request('https://api.imgur.com/3/image', 
+                                   method='POST', params={'image': image_file})
+
+            image_link = imgur_response["link"]
+
+            # Save photo URL with user
+            profile.profilePicture = image_link
+            profile.save()
+
+            request_response = json.dumps({"profileID": profile.accountID.accountID ,
+                                 "profilePicture": image_link})
+            
+            return JsonResponse(request_response, status = 201, safe = False)
+
+        except MultiValueDictKeyError:
+            return JsonResponse({"error": "Missing required JSON fields."}, status = 400)
+
+        except Profile.DoesNotExist:
+            return JsonResponse({"error": "Profile not found."}, status = 404)
+        
+        except HTTPError:
+            return JsonResponse({"error": "Uploaded file is not an image."}, status = 400)
+
+        except ValueError:
+            return JsonResponse({"error": "accountID or profilePicture field is empty."}, status = 400)
+
+    def patch(self, request, *args, **kwargs):
+
+        try:
+            data = MultiPartParser(request.META, request, request.upload_handlers).parse()
+            
+            account_id = data[0]["accountID"]
+
+            profile = Profile.objects.get(accountID=account_id)
+
+            if profile.profilePicture is None:
+                return JsonResponse({"error": "Profile does not have an existing profilePicture." +
+                                     "Use a POST request to upload a profilePicture."}, status = 409)
+            
+            uploaded_profile_picture = data[1]["profilePicture"]
+
+            imgur_client = initialize_imgur_client()
+            
+            # Grab raw photo from request files and encode it to upload to imgur
+            image_file = b64encode(uploaded_profile_picture.read())
+
+            # send POST request to imgur API to upload photo
+            imgur_response = imgur_client._send_request('https://api.imgur.com/3/image', 
+                                   method='POST', params={'image': image_file})
+
+            image_link = imgur_response["link"]
+
+            # Save photo URL with user
+            profile.profilePicture = image_link
+            profile.save()
+
+            request_response = json.dumps({"profileID": profile.accountID.accountID ,
+                                 "profilePicture": image_link})
+            
+            return JsonResponse(request_response, status = 201, safe = False)
```

Comment:
Shouldn't the status code be 200 since it's updating instead of creating a new resource?

Reply:
Yea that sounds right, I'll fix this!